### PR TITLE
feat(repo): migrate to pnpm

### PR DESCRIPTION
**Pull request by meeroslav**
This PR migrates our repo from `yarn classic` to `pnpm`:

- Migrate lock file
- Migrate CI and Nightly configs
- Add missing explicitly used packages
- Fix invalid imports
- Fix local scripts
- Fix broken tests

Current status:
- [X] Lint, Build and Test are passing
- [X]  CI and nightly are running properly
- [x] One `e2e-js` `swc` test is failing and several `e2e-react-native` tests are failing

## Related Issue(s)

Fixes #

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -148,6 +143,7 @@ jobs:
       NX_E2E_RUN_CYPRESS: 'true'
       NX_VERBOSE_LOGGING: 'false'
       NX_PERF_LOGGING: 'false'
+      NX_E2E_SKIP_BUILD_CLEANUP: 'true'
```
Comment (FrozenPandaz) [inline .circleci/config.yml:146]: Let's try to do this separately.

File: .circleci/config.yml [hunk 2]
```diff
@@ -175,6 +171,7 @@ jobs:
       NX_VERBOSE_LOGGING: 'false'
       NX_DAEMON: 'true'
       NX_PERF_LOGGING: 'false'
+      NX_E2E_SKIP_BUILD_CLEANUP: 'true'
```
Comment (FrozenPandaz) [inline .circleci/config.yml:174]: Let's try this in a separate PR.

File: .circleci/config.yml [hunk 3]
```diff
@@ -221,6 +218,7 @@ jobs:
       NX_E2E_CI_CACHE_KEY: e2e-circleci-macos
       NX_DAEMON: 'true'
       NX_PERF_LOGGING: 'false'
+      NX_E2E_SKIP_BUILD_CLEANUP: 'true'
```
Comment (FrozenPandaz) [inline .circleci/config.yml:221]: Let's try this in a separate PR.

File: .npmrc [hunk 4]
```diff
@@ -0,0 +1,4 @@
+strict-peer-dependencies=false
+auto-install-peers=true
+ignore-scripts=true # needed to avoid postinstall on nested nx
```
Comment (FrozenPandaz) [inline .npmrc:3]: We shouldn't need this...

File: .npmrc [hunk 5]
```diff
@@ -0,0 +1,4 @@
+strict-peer-dependencies=false
+auto-install-peers=true
```
Comment (FrozenPandaz) [inline .npmrc:2]: We should remove this.

File: packages/nx/package.json [hunk 6]
```diff
@@ -68,7 +68,7 @@
   },
   "peerDependencies": {
     "@swc-node/register": "^1.4.2",
-    "@swc/core": "^1.2.173"
+    "@swc/core": "1.3.11"
```
Comment (FrozenPandaz) [inline packages/nx/package.json:71]: Why did this change?

File: packages/nx/src/utils/register.ts [hunk 7]
```diff
@@ -51,12 +51,14 @@ export function registerTranspiler(
 
   const preferTsNode = process.env.NX_PREFER_TS_NODE === 'true';
 
+  console.log('AÀasfasfasfasfasfasf');
```
Comment (FrozenPandaz) [inline packages/nx/src/utils/register.ts:54]: rm

File: packages/nx/src/utils/register.ts [hunk 8]
```diff
@@ -51,12 +51,14 @@ export function registerTranspiler(
...
+
   if (swcNodeInstalled && !preferTsNode) {
     // These are requires to prevent it from registering when it shouldn't
     const { register } =
       require('@swc-node/register/register') as typeof import('@swc-node/register/register');
 
-    registerTranspiler = () => register(compilerOptions);
+    registerTranspiler = () => () => register(compilerOptions);
```
Comment (FrozenPandaz) [inline packages/nx/src/utils/register.ts:61]: Revert

File: packages/nx/src/utils/versions.ts [hunk 9]
```diff
@@ -6,5 +6,5 @@ export const prettierVersion = '^2.6.2';
 export const typescriptESLintVersion = '^5.58.0';
 export const eslintVersion = '~8.15.0';
 export const eslintConfigPrettierVersion = '8.1.0';
-export const swcNodeVersion = '^1.4.2';
+export const swcNodeVersion = '^1.5.4';
```
Comment (FrozenPandaz) [inline packages/nx/src/utils/versions.ts:9]: This shouldn't change here?

File: packages/workspace/src/generators/ci-workflow/ci-workflow.spec.ts [hunk 10]
```diff
@@ -8,7 +8,7 @@ import {
 import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
 import { ciWorkflowGenerator } from './ci-workflow';
 
-describe('lib', () => {
+describe('CI Workflow generator', () => {
   let tree: Tree;
 
   beforeEach(() => {
```
Comment (FrozenPandaz) [inline packages/workspace/src/generators/ci-workflow/ci-workflow.spec.ts:14]: Add tests for each package manager

File: pnpm-workspace.yaml [hunk 11]
```diff
@@ -0,0 +1,2 @@
+packages:
+  - packages/*
```
Comment (FrozenPandaz) [inline pnpm-workspace.yaml:2]: This seems like a larger change than we need here

File: scripts/check-imports.js [hunk 12]
```diff
@@ -15,7 +15,7 @@ function allFilesInDir(dirName) {
             name: child,
             content: fs.readFileSync(child).toString(),
           });
-        } else if (s.isDirectory()) {
+        } else if (s.isDirectory() && !child.includes('node_modules')) {
```
Comment (FrozenPandaz) [inline scripts/check-imports.js:18]: We shouldn't have `node_modules` within project directories.

File: scripts/copy-local-native.js [hunk 13]
```diff
@@ -6,6 +6,8 @@ const p = process.argv[2];
 
 const nativeFiles = glob.sync(`packages/${p}/**/*.node`);
 
-nativeFiles.forEach((file) => {
-  fs.copyFileSync(file, `build/${file}`);
-});
+nativeFiles
+  .filter((file) => !file.includes('/node_modules/'))
```
Comment (FrozenPandaz) [inline scripts/copy-local-native.js:10]: We shouldn't have `node_modules` within project directories.

File: scripts/depcheck/discrepancies.ts [hunk 14]
```diff
@@ -6,6 +6,7 @@ const IGNORE_MATCHES = {
   '*': [] as string[],
   angular: ['webpack-merge', '@phenomnomnominal/tsquery'],
   cypress: ['webpack', '@babel/core', 'babel-loader'],
+  jest: ['@jest/reporters', 'jest-config', 'jest-resolve', 'jest-util'],
```
Comment (FrozenPandaz) [inline scripts/depcheck/discrepancies.ts:9]: This should be resolved.

File: scripts/nx-release.ts [hunk 15]
```diff
@@ -83,7 +83,7 @@ function hideFromGitIndex(uncommittedFiles: string[]) {
     });
   }
 
-  const versionOptions = {
+  const versionOptions: Record<string, any> = {
```
Comment (FrozenPandaz) [inline scripts/nx-release.ts:86]: This shouldn't have changed?

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

**Review by mandarini — APPROVED** [review]

File: e2e/js/src/js-swc.test.ts [hunk 16]
```diff
@@ -134,6 +134,8 @@ export function x() {
 
     // update .swcrc to use path mappings
     updateJson(`libs/${lib}/.swcrc`, (json) => {
+      console.log(json);
```
Comment (nartc) [inline e2e/js/src/js-swc.test.ts:137]: nit: remove console log if not intentional

**Review by nartc — APPROVED** [review]

**Review by jaysoo — APPROVED** [review]

**Review by bcabanes — APPROVED** [review]

File: .circleci/config.yml [hunk 17]
```diff
@@ -221,6 +218,7 @@ jobs:
       NX_E2E_CI_CACHE_KEY: e2e-circleci-macos
       NX_DAEMON: 'true'
       NX_PERF_LOGGING: 'false'
+      SELECTED_PM: 'npm' # use the previous runner
```
Comment (FrozenPandaz) [inline .circleci/config.yml:221]: Why is this `npm`?

Comment (meeroslav) [reply .circleci/config.yml:221]: We run the command (on line 236) with `npx nx ...` so it automatically picks up `npm`. This just makes it explicit.

Comment (meeroslav) [reply .circleci/config.yml:221]: We only run `macos` tests on `npm` (something to be changed for nightly) but not as a part of this PR

**Review by barbados-clemens — APPROVED** [review]

File: packages/react-native/package.json [hunk 18]
```diff
@@ -42,7 +42,7 @@
     "@nx/workspace": "file:../workspace"
   },
   "peerDependencies": {
-    "react-native": "^0.71.7"
+    "react-native": ">= 0.71.0"
```
Comment (meeroslav) [inline packages/react-native/package.json:45]: Fix from https://github.com/nrwl/nx/pull/16524 for `expo` e2e

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**